### PR TITLE
Bump Gardener to `v1.101.3`

### PR DESCRIPTION
**Pull request by Duciwuci**
**What this PR does / why we need it**:
Version bump to gardener `v1.101.3`

**Which issue(s) this PR fixes**:
Update gardener version.

### CODE REVIEW

**Comment by robinschneider** [comment]
Your description is wrong.
It is not bumped to `v1.99.4`

**Comment by Duciwuci** [comment]
@robinschneider I updated the description :) Thanks!

File: go.mod [hunk 1]
```diff
@@ -1,35 +1,35 @@
 module github.com/stackitcloud/gardener-extension-acl
 
-go 1.22.0
+go 1.22.4
 
-toolchain go1.22.5
+toolchain go1.22.8
 
 replace k8s.io/client-go => k8s.io/client-go v0.29.6
```
Comment (MichaelEischer) [inline go.mod:7]: Please try to replace the weird `client-go` version below and also bump it to v0.29.7

Comment (Duciwuci) [reply go.mod]: I bumped `client-go` to `v0.29.7`.

File: go.mod [hunk 2]
```diff
@@ -1,35 +1,35 @@
 module github.com/stackitcloud/gardener-extension-acl
 
-go 1.22.0
+go 1.22.4
 
-toolchain go1.22.5
+toolchain go1.22.8
 
-replace k8s.io/client-go => k8s.io/client-go v0.29.6
+replace k8s.io/client-go => k8s.io/client-go v0.29.7
```
Comment (MichaelEischer) [inline go.mod:7]: The replace statement still should be removed

**Review by MichaelEischer — COMMENTED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]

**Pull request merged**